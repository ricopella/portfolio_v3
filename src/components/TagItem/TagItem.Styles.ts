import styled from '@emotion/styled'
import { COLORS } from '../../styles/variables'

const ItemWrapper = styled.div`
  display: grid;
  border: 0.3125rem solid transparent;
  grid-template-columns: max-content;
  grid-template-rows: max-content;
  max-width: 100%;
`

const Item = styled.div`
  background-color: ${COLORS.palette.mediumSlateBlue};
  border-radius: 0.3125rem;
  color: ${COLORS.palette.eerieBlack};
  font-size: 0.825rem;
  padding: 0.3125rem 0.3125rem;
  width: max-content;
`

export default {
  Item,
  ItemWrapper,
}
