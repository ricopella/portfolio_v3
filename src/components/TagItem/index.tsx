import React, { FC } from 'react'
import Styled from './TagItem.Styles'
import { TagItemProps } from '../../types'

const TagItem: FC<TagItemProps> = ({ title }) => (
  <Styled.ItemWrapper>
    <Styled.Item>{title}</Styled.Item>
  </Styled.ItemWrapper>
)

export default TagItem
