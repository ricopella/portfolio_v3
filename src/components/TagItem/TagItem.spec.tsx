import React from 'react'
import renderer from 'react-test-renderer'
import TagItem from './index'
import { render } from '@testing-library/react'

describe('<TagItem />', () => {
  it('should match TagItem snapshot', () => {
    const internalTree = renderer
      .create(<TagItem title="Wu-Tang Clan" />)
      .toJSON()

    expect(internalTree).toMatchSnapshot()
  })

  it('should render children', () => {
    const { getByText } = render(<TagItem title="Wu-Tang Clan" />)

    expect(getByText(/Wu-Tang Clan/i)).toBeTruthy()
  })
})
