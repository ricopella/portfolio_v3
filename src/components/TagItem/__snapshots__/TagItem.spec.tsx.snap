// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TagItem /> should match TagItem snapshot 1`] = `
<div
  className="css-3m0mjb"
>
  <div
    className="css-1t1bufi"
  >
    Wu-Tang Clan
  </div>
</div>
`;
